File: PDS_JAVA/PDS_WebAPI/src/main/java/com/demo/service/IActivityService.java
package com.demo.service;
import java.util.List;

import org.springframework.stereotype.Service;

import com.demo.model.Activity;

@Service
public interface IActivityService {

	List<Activity> getallactivities();

	void addnewactivity(Activity p);

	Activity getById(int pid);

	void updateById(Activity activity);

	void deleteById(int id);

	List<Activity> getByPrice(int lpr, int hpr);
	
}

File: PDS_DOTNET/PDSWebAPI/Service/IActivityService.cs
using System.Collections.Generic;
using System.Linq;
using Demo.Models; // Assuming Activity class is defined in Models namespace
using Microsoft.EntityFrameworkCore;

namespace Demo.Services
{
    public interface IActivityService
    {
        IEnumerable<Activity> GetAllActivities();
        Activity GetById(int id);
        void AddNewActivity(Activity activity);
        void UpdateById(Activity activity);
        void DeleteById(int id);
    }
}    